credoai/modules/model_modules/security.py

```diff
@@ -0,0 +1,145 @@
+import copy
+import warnings
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+from art.attacks.extraction import CopycatCNN
+from art.estimators.classification import KerasClassifier
+from art.estimators.classification.scikitlearn import SklearnClassifier
+from credoai.modules.credo_module import CredoModule
+from credoai.utils.common import NotRunError
+from keras.layers import Dense
+from keras.models import Sequential
+from keras.utils.np_utils import to_categorical
+from sklearn import metrics as sk_metrics
+
+tf.compat.v1.disable_eager_execution()
+import os
+os.environ['TF_CPP_MIN_LOG_LEVEL'] = '3' 
+
+class SecurityModule(CredoModule):
+    """Security module for Credo AI.
+
+    This module takes in binary classification model and data and 
+     provides functionality to perform security assessment
+
+    Parameters
+    ----------
+    model : model
+        A trained ML model
+    x_test : pandas.DataFrame
+        The test features
+    y_test : pandas.Series
+        The test outcome labels
+    """
+
+    def __init__(
+        self, model, x_test, y_test
+    ):
+        self.x_test = x_test.to_numpy()
+        self.y_test = to_categorical(y_test, num_classes=2)
```

Comment:
Couldn't most of this functionality work for multiclass classification? Why are we limiting it to binary classification rather than inferring the number of classes?

Reply:
I feel positive that it could. I go through the code to see what needs to be changed and upgrade the module accordingly.

Reply:
Cool, let's make a follow-up ticket quickly to extend this module

Reply:
Ticket created in the backlog.